### PR TITLE
XML2RFC Tests

### PR DESCRIPTION
**Pull request by stefanomunarini**

### CODE REVIEW

**Review by strogonoff — CHANGES_REQUESTED** [review]
Commit that changes “able” to “unable” should be reverted, the rest can be as is but let’s discuss questions raised in the comments

File: xml2rfc_compat/serializer.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def to_xml(item: BibliographicItem, anchor=None) -> Element:
 
     else:
         raise ValueError(
-            "Able to construct neither <reference> nor <referencegroup>: "
+            "Unable to construct neither <reference> nor <referencegroup>: "
```
[the remark this answers is not in the document]
Comment (stefanomunarini) [reply xml2rfc_compat/serializer.py:80]: Shall it be "Unable to construct either .. or .. ", then? 

To me "to be unable to do something" sounds better than "to be able to not do something".

[the remark this answers is not in the document]
Comment (strogonoff) [reply xml2rfc_compat/serializer.py:80]: Unless there is a grammatical mistake or a typo (which I believe there is not), let’s better set it aside and focus on the tests. If we start looking into improving phrasing (as opposed to straight up typos), there is a bottomless pit of work that is secondary to current objective, so I don’t want to create a precedent.

There could be an opportunity to address this stuff in future. Even for typos I would recommend noting them and postponing for inclusion in a separate PR later.

**Comment by strogonoff** [comment]
Discussed https://github.com/ietf-ribose/bibxml-service/pull/154/files#r832316535 1:1.

File: xml2rfc_compat/tests.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                          <xs:element type="frontType" name="front"/>
+                        </xs:sequence>
+                        <xs:attribute type="xs:string" name="anchor"/>
+                      </xs:complexType>
+                    </xs:schema>
+                """
+        )
+        xmlschema_doc = etree.parse(bibitem_xsd)
+        xmlschema = etree.XMLSchema(xmlschema_doc)
+        xmlschema.assertValid(xml)  # Validate XML Schema
```
[the remark this answers is not in the document]
Comment (strogonoff) [reply xml2rfc_compat/tests.py:122]: Filed in https://github.com/ietf-ribose/bibxml-service/issues/155

**Pull request merged**